### PR TITLE
Wurstified priest healing spells + Wurst lights gate

### PR DESCRIPTION
**Pull request by Marsunpaisti**
Custom-coded the priest heals so they use the HealingSystem. This lets us add modifiers to them if we want. Also made HealingSystem modifications stack multiplicatively.
**These spells have not been added to priest spellbooks in the object editor yet**

Lights gate:
Almost direct copy-paste from Dark Portal. Broken spells removed from random spell list. I suggest we also remove the special effects that sage spells have when they are cast inside the Light's Gate because most of them didnt work in the first place and sage is op as fuck anyway.

### CODE REVIEW

File: wurst/systems/spells/priest/HealingWave.wurst [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+    construct(unit caster, unit target, real healingInitialAmount, int bounces)
+        this.caster = caster
+        this.healingAmount = healingInitialAmount
+        this.bouncesLeft = bounces
+        this.owner = caster.getOwner()
+        waveBounce(target, caster, true)
+
+    function waveBounce(unit target, unit previousTarget, bool isFirstBounce)
+        var lightningFxPath = LIGHTNING_HEALING_WAWE_SEECONDARY
```
Comment (jonathanmclaus) [inline wurst/systems/spells/priest/HealingWave.wurst]: `SEECONDARY` has a typo. Also a good use case for the ternary operator.

Comment (Marsunpaisti) [reply wurst/systems/spells/priest/HealingWave.wurst]: I didnt make the constant. Swapping the operator though.

File: wurst/systems/spells/priest/RangedHeal.wurst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    new HealingInstance(target, 50, HealingType.ABILITY)
+    let fx = target.addEffect(HEAL_HIT_FX, "origin")
+    doAfter(1.5) ->
+        fx.destr()
+
+init
+    registerSpellEffectEvent(ABILITY_ID) ->
+        onCast()
+
+    /*
```
Comment (jonathanmclaus) [inline wurst/systems/spells/priest/RangedHeal.wurst]: Don't commit commented code.

Comment (Marsunpaisti) [reply wurst/systems/spells/priest/RangedHeal.wurst]: Fixd

File: wurst/systems/spells/priest/IncreaseMetabolism.wurst [hunk 3]
```diff
@@ -46,4 +46,4 @@ init
     onUnitHealed() ->
         let instance = getHealingInstance()
         if instance.target.hasAbility(BUFF_ID)
-            instance.increaseMultiplier(0.5)
+            instance.scaleMultiplier(1.5)
```
Comment (jonathanmclaus) [inline wurst/systems/spells/priest/IncreaseMetabolism.wurst]: Does this affect all healing or only item healing?

Comment (Marsunpaisti) [reply wurst/systems/spells/priest/IncreaseMetabolism.wurst]: Fixed

**Pull request merged**